src/main/java/org/opentripplanner/netex/mapping/FlexStopsMapper.java

```diff
@@ -119,17 +123,23 @@ AreaStop mapFlexArea(FlexibleArea area, Geometry geometry, String backupName) {
   }
 
   /**
-   * Allows pickup / drop off at any regular Stop inside the area
+   * Allows pickup / drop off at all regular stops inside the area that match the transit mode of the
+   * flexible stop place.
    */
-  List<RegularStop> findStopsInFlexArea(FlexibleArea area, Geometry geometry) {
+  List<RegularStop> findStopsInFlexArea(
+    FlexibleArea area,
+    Geometry geometry,
+    TransitMode flexibleStopTransitMode
```

Comment:
The mode and the geometry are nullable, isn't it? Can you add the appropriate annotations?